internal/prober/http/http_test.go

```diff
@@ -13,6 +13,73 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestName(t *testing.T) {
+	name := Prober.Name(Prober{})
+	require.Equal(t, name, "http")
+}
+
+func TestNewProber(t *testing.T) {
+	testcases := map[string]struct {
+		input    sm.Check
+		expected Prober
+	}{
+		"default": {
+			input: sm.Check{
+				Target: "www.grafana.com",
+				Settings: sm.CheckSettings{
+					Http: &sm.HttpSettings{},
+				},
+			},
+			expected: Prober{
+				config: config.Module{
+					Prober:  "http",
+					Timeout: 0,
+					HTTP: config.HTTPProbe{
+						ValidStatusCodes:   []int{},
+						ValidHTTPVersions:  []string{},
+						IPProtocol:         "ip6",
+						IPProtocolFallback: true,
+						Method:             "GET",
+						Headers: map[string]string{
+							"user-agent": version.UserAgent(),
+						},
+						FailIfBodyMatchesRegexp:      []config.Regexp{},
+						FailIfBodyNotMatchesRegexp:   []config.Regexp{},
+						FailIfHeaderMatchesRegexp:    []config.HeaderMatch{},
+						FailIfHeaderNotMatchesRegexp: []config.HeaderMatch{},
+						HTTPClientConfig: httpConfig.HTTPClientConfig{
+							FollowRedirects: true,
+						},
+					},
+				},
+			},
+		},
+		"no-settings": {
+			input: sm.Check{
+				Target: "www.grafana.com",
+				Settings: sm.CheckSettings{
+					Http: nil,
+				},
+			},
+			expected: Prober{},
+		},
+	}
+
+	for name, testcase := range testcases {
+		ctx := context.TODO()
```

Comment:
Conventionally `context.TODO()` means "I need a context here, I don't know what to do use right now, I will add one later". It's a sort of TODO note that is part of the code.

Usually you would use either contxt.Background() or context.WithTimeout, but for this particular test a timeout doesn't make sense (because the prober doesn't run).